security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -220,8 +221,9 @@ func (s *sdsservice) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecre
 			}
 
 			if discReq.Node == nil {
-				sdsServiceLog.Errorf("Close connection. Invalid discovery request with no node")
-				return fmt.Errorf("invalid discovery request with no node")
+				discReq.Node = node
```

Comment:
Before passing node to discReq.Node, can we check if node is initialized from previous discReq.Node already? If node is not initialized and discReq.Node is nil, then we still need an error message and close connection here.

Reply:
Added a new error check that the node must not be nil.

Reply:
Thanks for the change.